low-level/files/opt/seagate/sspl/bin/sspl_test

```diff
@@ -1,35 +1,89 @@
-
-plan=${1:-}
-avoid_rmq=${2:-}
-
-if rpm -qa | grep -q sspl-test ;
-then
-    $TEST_DIR/run_tests.sh test $plan $avoid_rmq
-    exit $?
-else
-    echo "Please install sspl-test RPM"
-    exit 1
-fi
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import subprocess
+
+# Add the top level directories
+sys.path.insert(0, '/opt/seagate/cortx/sspl/low-level')
+from framework.base.sspl_constants import PRODUCT_FAMILY
+
+TEST_DIR = f"/opt/seagate/{PRODUCT_FAMILY}/sspl/sspl_test"
+
+class Cmd:
+
+    def __init__(self, args: dict):
+        self._args = args.args
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @staticmethod
+    def add_args(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+
+        parser = argparse.ArgumentParser(desc)
+        parser.add_argument('args', nargs='*', default=[], help='args')
+        args = parser.parse_args(argv)
+        return args
+
+
+class SSPLTestCmd(Cmd):
+    "Run SSPL Test"
+    name = "sspl-test"
+
+    def __init__(self, args):
+        super().__init__(args)
+    
+    def process(self):
+        output=None
+        print("Processing %s %s" %(self.name, self._args))
+        CMD="rpm -qa | grep sspl-test"
+        output=subprocess.check_output(CMD, shell=True)
+        if output:
+            CMD=f"{TEST_DIR}/run_tests.sh test " + ' '.join(self._args)
+            process = subprocess.Popen(CMD, shell=True, stdout=subprocess.PIPE)
+            while True:
+                output = process.stdout.readline()
+                if output == '' and process.poll() is not None:
+                    break
+                if output:
+                    print(output.strip())
+            rc = process.poll()
+            return rc
+        else:
+            print("Please install sspl-test RPM")
+            sys.exit(1)
+
+
+def main(argv: dict):
+    try:
+        desc = "Run SSPL Test"
+        command = Cmd.add_args(desc, argv[1:])
+        SSPLTestCmd(command).process()
+    except Exception as e:
+        sys.stderr.write("error: %s\n\n" % str(e))
+        sys.stderr.write("%s\n" % traceback.format_exc())
```

Comment:
traceback in all setup scripts may add too much logs. Its being introduced in sspl_setup.py already, so keep it commented here, we will uncomment & use if needed

Reply:
Done